cmd/swapcli/main.go

```diff
@@ -620,33 +619,56 @@ func runGetPastSwapIDs(ctx *cli.Context) error {
 	}
 
 	fmt.Println("Past swap offer IDs:")
-	for i, id := range ids {
-		fmt.Printf("%d: %s\n", i, id)
-	}
 	if len(ids) == 0 {
 		fmt.Println("[none]")
+		return nil
 	}
+
+	for i, id := range ids {
+		if i > 0 {
+			fmt.Printf("---\n")
+		}
+
+		fmt.Printf("ID: %s\n", id.ID)
+		fmt.Printf("Start time: %s\n", id.StartTime)
+		fmt.Printf("End time: %s\n", id.EndTime)
+	}
+
 	return nil
 }
 
 func runGetOngoingSwap(ctx *cli.Context) error {
 	offerID := ctx.String(flagOfferID)
 
 	c := newRRPClient(ctx)
-	info, err := c.GetOngoingSwap(offerID)
+	resp, err := c.GetOngoingSwap(offerID)
 	if err != nil {
 		return err
 	}
 
-	receivedCoin := "ETH"
-	if info.Provided == coins.ProvidesETH {
-		receivedCoin = "XMR"
+	fmt.Println("Ongoing swaps:")
+	if len(resp.Swaps) == 0 {
+		fmt.Println("[none]")
+		return nil
 	}
 
-	fmt.Printf("Provided: %s %s\n", info.ProvidedAmount.Text('f'), info.Provided)
-	fmt.Printf("Receiving: %s %s\n", info.ExpectedAmount, receivedCoin)
-	fmt.Printf("Exchange Rate: %s ETH/XMR\n", info.ExchangeRate)
-	fmt.Printf("Status: %s\n", info.Status)
+	for i, info := range resp.Swaps {
+		if i > 0 {
+			fmt.Printf("---\n")
+		}
+
+		receivedCoin := "ETH"
+		if info.Provided == coins.ProvidesETH {
+			receivedCoin = "XMR"
+		}
+
+		fmt.Printf("ID: %s\n", info.ID)
+		fmt.Printf("Start time: %s\n", info.StartTime)
+		fmt.Printf("Provided: %s %s\n", info.ProvidedAmount.Text('f'), info.Provided)
+		fmt.Printf("Receiving: %s %s\n", info.ExpectedAmount, receivedCoin)
```

Comment:
`.Text('f')` if you want to guarantee that exponent notation never gets used. The exchange rate below doesn't need it, as it is our own type.

Reply:
Same thing for the identical line in `runGetPastSwap`.

Reply:
It looks like you removed the .Text('f') call on the `ProvidedAmount` instead of adding it to the `ExpectedAmount`. I was suggesting that we add it to both. Were you thinking that exponent notation is fine?

I'm playing around to see when `apd.Decimal` adds the exponent notation. At least in this example, it switches over the exponent notation at 7 decimals:

```
func TestExample(t *testing.T) {
	amt := coins.StrToDecimal("0.000001")
	require.Equal(t, "0.000001", amt.String())

	amt = coins.StrToDecimal("0.0000001")
	require.Equal(t, "1E-7", amt.String())
}
```